plugins/modules/database/misc/odbc.py

```diff
@@ -0,0 +1,157 @@
+      type: str
+    query:
+      description:
+        - The SQL query to perform
+      required: yes
+      type: str
+    params:
+      description:
+        - Parameters to pass to the SQL query
+      type: list
+      elements: str
+
+requirements:
+  - "python >= 2.6"
+  - "pyodbc"
+
+notes:
+  - "Like the command module, this module always returns changed = yes whether or not the query would change the database."
+  - "To alter this behavior you can use C(changed_when): [yes or no]."
+  - "For details about return values (description and row_count) see https://github.com/mkleehammer/pyodbc/wiki/Cursor."
+'''
+
+EXAMPLES = '''
+- name: Set some values in the test db
+  odbc:
+    dsn: "DRIVER={ODBC Driver 13 for SQL Server};Server=db.ansible.com;Database=my_db;UID=admin;PWD=password;"
+    query: "Select * from table_a where column1 = ?"
+    params:
+      - "value1"
+  changed_when: no
+'''
+
+RETURN = '''
+results:
+    description: List of dicts containing selected rows, likely empty for DDL statements.
+    returned: success
+    type: list
+    elements: str
+description:
+    description: "List of dicts about the columns selected from the cursors, likely emptry for DDL statements. See notes."
+    returned: success
+    type: list
+    elements: str
+row_count:
+    description: "The number of rows selected or modified according to the cursor defaults to -1. See notes."
+    returned: success
+    type: str
+'''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native
+
+HAS_PYODBC = None
+try:
+    import pyodbc
+    HAS_PYODBC = True
+except ImportError as e:
+    HAS_PYODBC = False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            dsn=dict(type='str', required=True, no_log=True),
+            query=dict(type='str', required=True),
+            params=dict(type='list', elements='str'),
+        ),
+    )
+
+    dsn = module.params.get('dsn')
+    query = module.params.get('query')
+    params = module.params.get('params')
+
+    if not HAS_PYODBC:
+        module.fail_json(msg=missing_required_lib('pyodbc'))
+
+    # Try to make a connection with the DSN
+    connection = None
+    try:
+        connection = pyodbc.connect(dsn)
+    except Exception as e:
+        module.fail_json(msg='Failed to connect to DSN: {0}'.format(to_native(e)))
+
+    result = dict(
+        changed=True,
+        description=[],
+        row_count=-1,
+        results=[],
+    )
+
+    try:
+        cursor = connection.cursor()
+
+        if params:
+            cursor.execute(query, params)
+        else:
+            cursor.execute(query)
+        cursor.commit()
+        try:
+            # Get the rows out into an 2d array
+            for row in cursor.fetchall():
+                new_row = []
+                for column in row:
+                    new_row.append("{0}".format(column))
+                result['results'].append(new_row)
```

Comment:
`result['results']` seems to be a list of **lists**, not of dicts or strings.

Reply:
Is there a way to indicate a list of lists of string? Ir us the best we can do is do get to list of lists?

Reply:
No, unfortunately there isn't. List of lists is the best you can do right now.